src/table/table-role/grid-navigation.tsx

```diff
@@ -168,74 +183,50 @@ class GridNavigationHelper {
+        return this.moveFocusBy(from, { y: 0, x: -1 });
 
       case KeyCode.right:
         event.preventDefault();
-        return moveFocusBy(this.table, from, { y: 0, x: 1 });
+        return this.moveFocusBy(from, { y: 0, x: 1 });
 
       case KeyCode.pageUp:
         event.preventDefault();
-        return moveFocusBy(this.table, from, { y: -this.pageSize, x: 0 });
+        return this.moveFocusBy(from, { y: -this.pageSize, x: 0 });
 
       case KeyCode.pageDown:
         event.preventDefault();
-        return moveFocusBy(this.table, from, { y: this.pageSize, x: 0 });
+        return this.moveFocusBy(from, { y: this.pageSize, x: 0 });
 
       case KeyCode.home:
         event.preventDefault();
-        return moveFocusBy(this.table, from, { y: 0, x: minExtreme });
+        return this.moveFocusBy(from, { y: 0, x: minExtreme });
 
       case KeyCode.end:
         event.preventDefault();
-        return moveFocusBy(this.table, from, { y: 0, x: maxExtreme });
+        return this.moveFocusBy(from, { y: 0, x: maxExtreme });
 
       case -KeyCode.home:
         event.preventDefault();
-        return moveFocusBy(this.table, from, { y: minExtreme, x: minExtreme });
+        return this.moveFocusBy(from, { y: minExtreme, x: minExtreme });
 
       case -KeyCode.end:
         event.preventDefault();
-        return moveFocusBy(this.table, from, { y: maxExtreme, x: maxExtreme });
+        return this.moveFocusBy(from, { y: maxExtreme, x: maxExtreme });
 
       default:
         return;
     }
   };
 
-  private onTableNodeMutation = (mutationRecords: MutationRecord[]) => {
-    // When focused cell is un-mounted the focusout event handler removes this.cell,
-    // while this.prevFocusedCell is retained until the next focusin event.
-    const cell = this.focusedCell ?? this.prevFocusedCell;
-    const cellSuppressed = cell ? this.isSuppressed(cell.element) : false;
-
-    // Update table elements focus if new nodes were added.
-    if (mutationRecords.some(record => record.addedNodes.length > 0)) {
-      muteElementFocusables(this.table, cellSuppressed);
-      ensureSingleFocusable(this.table, cell);
-    }
-
-    if (cell) {
-      for (const record of mutationRecords) {
-        if (record.type === 'childList') {
-          // The lost focus in an unmount event is reapplied to the table using the previous cell position.
-          // The moveFocusBy takes care of finding the closest position if the previous one no longer exists.
-          for (const removedNode of Array.from(record.removedNodes)) {
-            if (removedNode === cell.element || nodeContains(removedNode, cell.element)) {
-              ensureSingleFocusable(this.table, cell);
-              moveFocusBy(this.table, cell, { y: 0, x: 0 });
-            }
-          }
-        }
-      }
-    }
-  };
+  private moveFocusBy(cell: FocusedCell, delta: { x: number; y: number }) {
+    getNextFocusable(this.table, cell, delta)?.focus();
```

Comment:
Missing return statement?

Reply:
No, this one just focuses on the next applicable element. The case statements return to prevent multiple matches. The same can be achieved by using `break` after each case statement.